src/commands/generateFunctions.ts

```diff
@@ -0,0 +1,241 @@
+
+// tslint:disable: no-non-null-assertion
+export async function generateFunctions(node?: ApiTreeItem): Promise<void> {
+    if (!node) {
+        node = <ApiTreeItem>await ext.tree.showTreeItemPicker(ApiTreeItem.contextValue);
+    }
+
+    ext.outputChannel.show();
+
+    const openApiEditor: OpenApiEditor = new OpenApiEditor();
+    const openAPIDocString = await openApiEditor.getData(node);
+
+    if (!await validateAutorestInstalled()) {
+        return;
+    }
+
+    // tslint:disable-next-line: no-unsafe-any
+    const languages : string[] = Object.keys(languageTypes).map(key => languageTypes[key]);
+    const language = await ext.ui.showQuickPick(
+        languages.map((s) => {return { label: s, description: '', detail: '' }; }), { placeHolder: "Select language", canPickMany: false});
+
+    if (!await checkEnvironmentInstalled(language.label)) {
+        throw new Error(`You haven't installed '${language.label}' on your machine, please install '${language.label}' to continue.`);
+    }
+
+    let namespace = "";
+    if (language.label === languageTypes.Java) {
+        namespace = await askJavaNamespace();
+    } else if (language.label === languageTypes.CSharp) {
+        namespace = await askCSharpNamespace();
+    }
+
+    const uris = await askFolder();
+
+    const openAPIFilePath = path.join(uris[0].fsPath, `${node!.apiContract.name}.json`);
+
+    await window.withProgress(
+        {
+            location: ProgressLocation.Notification,
+            title: localize("openAPI", `Downloading OpenAPI Document for API '${node.apiContract.name}'...`),
+            cancellable: false
+        },
+        async () => {
+            await fse.writeFile(openAPIFilePath, openAPIDocString);
+        }
+    ).then(async () => {
+        window.showInformationMessage(localize("openAPIDownloaded", `Downloaded OpenAPI Document for API '${node!.apiContract.name} successfully.`));
+    });
+
+    await window.withProgress(
+        {
+            location: ProgressLocation.Notification,
+            title: localize("generateFunctions", `Generating functions for API '${node.apiContract.name}'...`),
+            cancellable: false
+        },
+        async () => {
+            const args: string[] = [];
+            args.push(`--input-file:${openAPIFilePath} --version:3.0.6314`);
+            args.push(`--output-folder:${uris[0].fsPath}`);
```

Comment:
Also - `cpUtils.wrapArgInQuotes()` in here.
```suggestion
            args.push(`--output-folder:${cpUtils.wrapArgInQuotes(uris[0].fsPath})`);
```